gateway/exporters/fluentd/fluentd.go

```diff
@@ -0,0 +1,229 @@
+	GnmiPath      string            `msg:"gnmiPath"`
+	SubscribePath string            `msg:"subscribedPath"`
+	Value         interface{}       `msg:"value"`
+	Meta          map[string]string `msg:"meta"`
+	FabricID      string            `msg:"fabricId"`
+	RackID        string            `msg:"rackId"`
+	DeviceID      string            `msg:"deviceId"`
+	ExtensionID   string            `msg:"extensionId"`
+	DeviceName    string            `msg:"deviceName"`
+	Timestamp     string            `msg:"gnmiTimestamp"`
+}
+
+func init() {
+	exporters.Register(Name, NewFluentdExporter)
+}
+
+func NewFluentdExporter(config *configuration.GatewayConfig) exporters.Exporter {
+	return &FluentdExporter{
+		config: config,
+	}
+}
+
+type FluentdExporter struct {
+	config       *configuration.GatewayConfig
+	connMgr      *connections.ConnectionManager
+	fluentLogger *fluent.Fluent
+}
+
+func (e *FluentdExporter) Name() string {
+	return Name
+}
+
+func (e *FluentdExporter) Export(leaf *ctree.Leaf) {
+	notification := leaf.Value().(*gnmipb.Notification)
+
+	for _, update := range notification.Update {
+		log := Log{}
+
+		value, valid := utils.GetValues(update.Val)
+
+		if !valid {
+			continue
+		}
+
+		log.Value = value
+
+		notificationPath := utils.GetTrimmedPath(notification.Prefix, update.Path)
+
+		elems, keys, err := extractPrefixAndPathKeys(notification.GetPrefix(), update.GetPath())
+		if err != nil {
+			e.config.Log.Info().Msg(fmt.Sprintf("Failed to extract notificationPath or keys: %s", err))
+		}
+
+		if log.Event == "" {
+			subscribePath := strings.Join(elems, "/")
+
+			p := notification.GetPrefix()
+			target := p.GetTarget()
+			origin := p.GetOrigin()
+
+			log.SubscribePath = subscribePath
+			log.GnmiPath = getPathWithKeys(*update.Path)
+
+			if origin != "" {
+				keys["origin"] = origin
+			}
+
+			if target != "" {
+				keys["target"] = target
+			}
+
+			log.Event = notificationPathToMetricName(subscribePath)
+
+			log.Meta = keys
+
+			if e.connMgr != nil {
+				targetName := log.Meta["target"]
+				targetConfig, found := (*e.connMgr).GetTargetConfig(targetName)
+
+				if found {
+					deviceID, exists := targetConfig.Meta["deviceID"]
+					if exists {
+						log.DeviceID = deviceID
+						log.DeviceName = path.Base(deviceID)
+					} else {
+						e.config.Log.Error().Msg("Device ARM ID is not set in the metadata of target: " + targetName)
+						return
+					}
+
+					log.RackID, exists = targetConfig.Meta["rackID"]
+					if !exists {
+						e.config.Log.Error().Msg("Rack ARM ID is not set in the metadata of target: " + targetName)
+						return
+					}
+
+					log.FabricID, exists = targetConfig.Meta["fabricID"]
+					if !exists {
+						e.config.Log.Error().Msg("Fabric ARM ID is not set in the metadata of target: " + targetName)
+						return
+					}
+
+					log.ExtensionID = e.config.Exporters.ExtensionArmId
+
+					// Used for aditional metadata fields
+					for _, fieldName := range e.config.ExporterMetadataAllowlist {
```

Comment:
nit: if rackID, deviceID or fabricID are specified in ExporterMetadataAllowlist, they'll be included twice.